test/js/bun/test/expect.test.js

```diff
@@ -4671,6 +4671,25 @@ describe("expect()", () => {
   test(' " " to contain ""', () => {
     expect(" ").toContain("");
   });
+
+  test("should work #13267", () => {
+    try {
+      expect(() => {
+        throw "!";
+      }).not.toThrow(/ball/);
+    } catch (e) {
+      expect(e).toBeUndefined();
+    }
+    try {
+      expect(() => {
+        throw "ball";
```

Comment:
If this is number, jest will stringify it?

Reply:
Yeah it converts to string confirmed 
<img width="358" alt="Screenshot 2024-08-13 at 14 29 13" src="https://github.com/user-attachments/assets/71f3a7f2-1a47-401c-ab75-51142bf5fb07">